### PR TITLE
Allow configurable settlement contract and native token address

### PR DESCRIPTION
**Pull request by vkgnosis**
Needed for e2e tests.

### Test Plan

CI, new functionality is not yet used or tested

### CODE REVIEW

**Review by nlordell — APPROVED** [review]

**Comment by nlordell** [comment]
Do we do a `GPv2Settlement::deployed()` anywhere in code?

**Comment by vkgnosis** [comment]
We're not supposed to. Maybe a year ago I remember going through the code and removing all such instances because they were hard to maintain. I just checked by grepping for `::deployed` and it seems still fine. Only happens in tests.

File: crates/shared/src/arguments.rs [hunk 1]
```diff
@@ -427,6 +435,12 @@ impl Display for Arguments {
             self.balancer_pool_deny_list
         )?;
         display_secret_option(f, "solver_competition_auth", &self.solver_competition_auth)?;
+        display_option(
+            f,
+            "settlement_contract_address",
+            &self.settlement_contract_address,
+        )?;
+        display_option(f, "native_token_address", &self.native_token_address)?;
```
Comment (MartinquaXD) [inline crates/shared/src/arguments.rs:443]: Didn't `H160` need to use `Debug` formatting to get the complete readable address?

Comment (nlordell) [reply crates/shared/src/arguments.rs:443]: Nice catch!

Comment (vkgnosis) [reply crates/shared/src/arguments.rs]: fixed

**Review by MartinquaXD — APPROVED** [review]

File: crates/shared/src/arguments.rs [hunk 2]
```diff
@@ -292,6 +292,14 @@ pub struct Arguments {
     /// The number of pools to initially populate the UniswapV3 cache
     #[clap(long, env, default_value = "100")]
     pub max_pools_to_initialize_cache: u64,
+
+    /// Override address of the settlement contract.
+    #[clap(long, env)]
+    pub settlement_contract_address: Option<H160>,
+
+    /// Override address of the settlement contract.
```
Comment (sunce86) [inline crates/shared/src/arguments.rs:300]: Comment copy paste

**Pull request merged**